tests/feature-store/test_steps.py

```diff
@@ -56,3 +73,471 @@ def test_set_event_random_id():
+        "fs-new",
+        entities=[fs.Entity("id")],
+        description="feature set",
+        engine="pandas",
+    )
+    # Pre-processing grpah steps
+    data_set_pandas.graph.to(OneHotEncoder(mapping=one_hot_encoder_mapping))
+    data_set_pandas._run_db = rundb_mock
+
+    data_set_pandas.reload = unittest.mock.Mock()
+    data_set_pandas.save = unittest.mock.Mock()
+    data_set_pandas.purge_targets = unittest.mock.Mock()
+    # Create a temp directory:
+    output_path = tempfile.TemporaryDirectory(suffix=".parquet")
+
+    # Ingest our dataset through our defined pipeline
+    df_pandas = fs.ingest(
+        data_set_pandas, data, targets=[ParquetTarget(path=output_path.name)]
+    )
+
+    data_ref = pd.DataFrame(
+        {
+            "name": data["name"],
+            "age": data["age"],
+            "department_IT": [1, 0, 0, 0, 1],
+            "department_RD": [0, 1, 1, 0, 0],
+            "department_Marketing": [0, 0, 0, 1, 0],
+            "timestamp": [
+                time.time(),
+                time.time(),
+                time.time(),
+                time.time(),
+                time.time(),
+            ],
+        },
+        index=[0, 1, 2, 3, 4],
+    )
+
+    assert isinstance(df_pandas, pd.DataFrame)
+    pd.testing.assert_frame_equal(
+        df_pandas,
+        data_ref,
+        check_dtype=True,
+        check_index_type=True,
+        check_column_type=True,
+        check_like=True,
+        check_names=True,
+    )
+
+    # Define the corresponding FeatureSet
+    data_set = fs.FeatureSet(
+        "fs-new",
+        entities=[fs.Entity("id")],
+        description="feature set",
+    )
+    # Pre-processing grpah steps
+    data_set.graph.to(OneHotEncoder(mapping=one_hot_encoder_mapping))
+    data_set._run_db = rundb_mock
+
+    data_set.reload = unittest.mock.Mock()
+    data_set.save = unittest.mock.Mock()
+    data_set.purge_targets = unittest.mock.Mock()
+
+    # Ingest our dataset through our defined pipeline
+    df = fs.ingest(data_set, data, targets=[ParquetTarget(path=output_path.name)])
+    pd.testing.assert_frame_equal(
+        df,
+        df_pandas,
+        check_dtype=True,
+        check_index_type=True,
+        check_column_type=True,
+        check_like=True,
+        check_names=True,
+    )
+
+
+def test_pandas_step_imputer(rundb_mock):
+    data, data_ref = get_data(True)
+    # Define the corresponding FeatureSet
+    data_set_pandas = fs.FeatureSet(
+        "fs-new",
+        entities=[fs.Entity("id")],
+        description="feature set",
+        engine="pandas",
+    )
+    # Pre-processing grpah steps
+    data_set_pandas.graph.to(Imputer(mapping={"department": "IT"}))
+    data_set_pandas._run_db = rundb_mock
+
+    data_set_pandas.reload = unittest.mock.Mock()
+    data_set_pandas.save = unittest.mock.Mock()
+    data_set_pandas.purge_targets = unittest.mock.Mock()
+    # Create a temp directory:
+    output_path = tempfile.TemporaryDirectory(suffix=".parquet")
```

Comment:
I'm not sure I understand why the directory is suffixed with `.parquet`... This will confuse mlrun into thinking that the directory is a file.